crates/validator/src/txn_validator.rs

```
@@ -164,14 +165,13 @@ impl TxnValidator {
     // TODO, to be synchronized with transaction fees.
     pub fn validate_amount(
         &self,
-        account_state: &HashMap<Address, Account>,
+        state_reader: StateStoreReadHandle,
         txn: &TransactionKind,
     ) -> Result<()> {
         let address = txn.sender_address();
         if let Ok(address) = secp256k1::PublicKey::from_str(address.to_string().as_str()) {
-            let account = account_state
-                .get(&Address::new(address))
-                .ok_or(TxnValidatorError::SenderAddressIncorrect)?;
+            let account = state_reader.get(&Address::new(address))
+                .map_err(|_| TxnValidatorError::SenderAddressIncorrect)?;
             if (account.credits() - account.debits())
```

Minor: please report the original error.